lib/ts/recipe/session/types.ts

```diff
@@ -322,3 +418,77 @@ export type SessionInformation = {
     accessTokenPayload: any;
     timeCreated: number;
 };
+
+export type ClaimValidationResult = { isValid: true } | { isValid: false; reason?: JSONValue };
+export type ClaimValidationError = {
+    id: string;
+    reason?: JSONValue;
```

Comment:
I think we could allow users to return just a string (or an array, etc.) instead of an object in custom claims. I'm not sure if that's useful, but it allows for more flexibility.